x/ibc/07-tendermint/types/update.go

```diff
@@ -33,44 +35,45 @@ import (
 // the new latest height
 // Tendermint client validity checking uses the bisection algorithm described
 // in the [Tendermint spec](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client.md).
-func CheckValidityAndUpdateState(
-	clientState clientexported.ClientState, consState clientexported.ConsensusState,
-	header clientexported.Header, currentTimestamp time.Time,
+func (cs *ClientState) CheckValidityAndUpdateState(
+	ctx sdk.Context, cdc codec.BinaryMarshaler, clientStore sdk.KVStore,
+	header clientexported.Header,
 ) (clientexported.ClientState, clientexported.ConsensusState, error) {
-	tmClientState, ok := clientState.(*types.ClientState)
+	tmHeader, ok := header.(Header)
 	if !ok {
 		return nil, nil, sdkerrors.Wrapf(
-			clienttypes.ErrInvalidClientType, "expected type %T, got %T", types.ClientState{}, clientState,
+			clienttypes.ErrInvalidHeader, "expected type %T, got %T", Header{}, header,
 		)
 	}
 
-	tmConsState, ok := consState.(*types.ConsensusState)
-	if !ok {
+	// Get trusted consensus state and unmarshal from clientStore
+	consBytes := clientStore.Get(host.KeyConsensusState(tmHeader.TrustedHeight))
```

Comment:
use helper function `GetConsensusState`

Reply:
++

Reply:
I can't because I'm using the `prefix.Store` which only handles Get and Set, I think this is preferable to giving the specific clients access to the `client.Keeper`

Reply:
I'm confused? I was referring to the helper in `x/ibc/07-tendermint/types/store.go`

```
// GetConsensusState retrieves the consensus state from the client prefixed
// store. An error is returned if the consensus state does not exist.
func GetConsensusState(store sdk.KVStore, cdc codec.BinaryMarshaler, height uint64) (*ConsensusState, error) {
```
This isn't giving access to the keeper

Reply:
😮 I did not realize that function existed!! That's a lot cleaner! Thanks @colin-axner !!